mySqlObj.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+class mySqlObj {
+
+	private $mysqli;
+
+	public function __construct() {
+		//mysqliクラスのオブジェクトを作成
+		$this->mysqli = new mysqli('localhost', 'root', '', 'testbase');
+		if ($this->mysqli->connect_error){
+			print("接続失敗：" . $this->mysqli->connect_error);
+			exit();
+		}
+	}
+
+	public function selectUserCount() {		
+		//userinfoテーブル内の件数を取得
+		$all_count = $this->mysqli->query("SELECT COUNT(*) cnt  FROM userinfo");
+		if($all_count){
+			while($cont = $all_count->fetch_object()){
+				$cnt = htmlspecialchars($cont->cnt);
+				return $cnt;
+			}
+		}
+		var_dump($cnt);
+		// DBを閉じる
+		$this->mysqli->close();
+	}
+
+	public function selectBulletinBoard($now_page,$max_page) {
+		//userinfoテーブルから登録日付の降順でデータを取得
+		$result = $this->mysqli->query("SELECT * FROM userinfo ORDER BY created LIMIT 10");
+		if($result){
+			//1行ずつ取り出し
+			while($row = $result->fetch_object()){
+			//エスケープして表示
+			$id = htmlspecialchars($row->id);
+			$name = htmlspecialchars($row->name);
+			$message = htmlspecialchars($row->message);
+			$created = htmlspecialchars($row->created);
+			$userdate = array('id' => $id, 'message' => $message);
+			print("<b>$id: $name : </b><br>");
+			print("$message <br>");
+			print("($created)<br>");
+			echo '<a href="./updateUser.php?'.http_build_query($userdate, '', '&amp;').'">変更</a> ';
+			echo '<a href="testlist.php?page='.$now_page.'&maxpage='.$max_page.'&'.http_build_query($userdate, '', '&amp;').'&mode=del">削除</a><br>';
+			}
+		}
+		// DBを閉じる
+		$this->mysqli->close();
+	}
+
+	public function selectBulletinBoardNext($now_page,$max_page) {
+		//接続を呼ぶ
+		//$this->dbConn();
+		//userinfoテーブルから登録日付の降順でデータを取得
+		$offset = ($now_page * 10) - 10;
+		$result = $this->mysqli->query("SELECT * FROM userinfo ORDER BY created LIMIT 10 OFFSET ".$offset);
+		if($result){
+		  //1行ずつ取り出し
+		  while($row = $result->fetch_object()){
+			//エスケープして表示
+			$id = htmlspecialchars($row->id);
+			$name = htmlspecialchars($row->name);
+			$message = htmlspecialchars($row->message);
+			$created = htmlspecialchars($row->created);
+			$userdate = array('id' => $id, 'message' => $message);
+			print("<b>$id: $name : </b><br>");
+			print("$message <br>");
+			print("($created)<br>");
+			echo '<a href="./updateUser.php?'.http_build_query($userdate, '', '&amp;').'">変更</a>    ';
+			echo '<a href="testlist.php?page='.$now_page.'&maxpage='.$max_page.'&'.http_build_query($userdate, '', '&amp;').'&mode=del">削除</a><br>';
+		  }
+		}
+		// DBを閉じる
+		$this->mysqli->close();
+	}
+
+	public function selectChkId($id) {
+		$result = $this->mysqli->query("SELECT id FROM userinfo where id = ".$id."");
+		if($result){
+			//1行取り出し
+			while($row = $result->fetch_object()){
+				//エスケープして表示
+				$chk_id = htmlspecialchars($row->id);
+				return $chk_id;
+			}
+		}
+		// DBを閉じる
+		$this->mysqli->close();
+	}
+
+	public function insertUserInfo($name, $mailaddress, $comment) {
+		//オートコミットをOFF
+		$stmt = $this->mysqli->autocommit(false);
```

Comment:
try{
$this->mysqli->begin_transaction(MYSQLI_TRANS_START_READ_ONLY)
$this->mysqli->query("");
$this->mysqli->commit();
} catch (Exception $e) {
    $this->mysqli->->rollback();
    throw $e;
}